target/linux/mediatek/filogic/base-files/lib/upgrade/platform.sh

```diff
@@ -108,9 +108,12 @@ platform_do_upgrade() {
 		esac
 		;;
 	cmcc,rax3000m)
```

Comment:
Please merge this with the entry for 
```
        bananapi,bpi-r3|\
        bananapi,bpi-r3-mini|\
        bananapi,bpi-r4|\
        bananapi,bpi-r4-poe|\
        jdcloud,re-cp-03|\
        mediatek,mt7988a-rfb|\
        openwrt,one)
```
which already contains all needed for this device as well.

Reply:
Ah, ok, so the `default` case would catch SPI-NAND. Actually we can convert the common case covering all the other devices to also do that, ie.
```diff
--- a/target/linux/mediatek/filogic/base-files/lib/upgrade/platform.sh
+++ b/target/linux/mediatek/filogic/base-files/lib/upgrade/platform.sh
@@ -101,7 +101,7 @@ platform_do_upgrade() {
                        PART_NAME="/dev/mtd${bootdev:8}"
                        default_do_upgrade "$1"
                        ;;
-               ubiblock*)
+               *)
                        CI_KERNPART="fit"
                        nand_do_upgrade "$1"
                        ;;
```

This is possible because the problem described in https://github.com/openwrt/openwrt/pull/14993#issuecomment-2033345822 only exists for ubiblock devices which were not yet created. In case of eMMC or SPI-NOR the boot device can always be detected by `fitblk_get_bootdev` as the partitions *do* exist already.

Reply:
Thank you, updated.

Reply:
I came up with a solution to the problem discussed in https://github.com/openwrt/openwrt/pull/14993#issuecomment-2033345822
See
https://github.com/openwrt/openwrt/pull/15963